dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -105,43 +105,48 @@ def self.extract_roles_from_user_permissions
     query = get_extraction_query(source_sql, 'email', ['permission'], true, 'dashboard.user_permissions')
     ContactRollupsV2.execute_query_in_transaction(query)
   end
-  
-  def self.extract_user_geos
+
+  def self.extract_users_and_geos
     # An user can have many user_geos records. user_geos records starts with only NULL
     # values until a cronjob runs, does IP-to-address lookup, and update them later.
-    teacher_geos_query = <<~SQL
+    teacher_and_geo_query = <<~SQL
       SELECT
-        teachers.email,
-        user_geos.city, user_geos.state, user_geos.postal_code, user_geos.country,
-        MAX(user_geos.updated_at) as updated_at
-      FROM (#{teacher_query('id, email')}) AS teachers
-      JOIN user_geos
-      ON teachers.id = user_geos.user_id
-      GROUP BY email, city, state, postal_code, country
+        t.email, t.id as user_id,
+        ug.city, ug.state, ug.postal_code, ug.country,
+        MAX(GREATEST(t.updated_at, IFNULL(ug.updated_at, t.updated_at))) as updated_at
+      FROM (#{teacher_query('id, email, updated_at')}) AS t
+      LEFT OUTER JOIN user_geos AS ug
+      ON t.id = ug.user_id
+      GROUP BY email, t.id, city, state, postal_code, country
     SQL
 
     extraction_query = get_extraction_query(
-      teacher_geos_query,
+      teacher_and_geo_query,
       'email',
-      %w(city state postal_code country),
+      %w(user_id city state postal_code country),
       true,
-      'dashboard.user_geos'
+      'dashboard.users'
     )
     ContactRollupsV2.execute_query_in_transaction(extraction_query)
   end
 
   def self.extract_pd_enrollments
     enrollment_email_query = <<~SQL
-      SELECT email, MAX(updated_at) AS updated_at
-      FROM pd_enrollments
+      SELECT
+        e.email,
+        w.course,
+        MAX(GREATEST(e.updated_at, IFNULL(w.updated_at, e.updated_at))) AS updated_at
```

Comment:
There aren't any records where an enrollment's workshop doesn't exist, so not sure the IFNULL is necessary.

Reply:
Can a workshop with existing enrollments be deleted? I prefer to play if safe when dealing with data because data/table-relationships can change overtime.

Reply:
It looks like they just get soft deleted, so the rows still exist. I don't know exactly what that means for enrollment (ie, should we exclude soft deleted enrollments / workshops)? I think fine to leave as-is for now.